tools/split-test.py

```diff
@@ -0,0 +1,63 @@
+"""
+The unit test program for testing the split tracer. It performs one direct write to the device specified
+by users, with the size that is 8 times greater than the maximum bio sector size supported by the device. 
+The submitted bio should be split in 7 times, resulting in 7 lines of output records in the split tracer. 
+The test checks that the number of lines of split output produced by the process is 7. 
+
+It requires the path to the target device as input that the test writes to. It also requires the maximum 
+bio sector size (in KB) supported by the device (which can be checked via /sys/block/sdX/queue/max_sectors_kb).
+"""
+
+import argparse
+import os
+from subprocess import Popen, PIPE
+from signal import SIGINT
+from time import sleep
+
+
+def unit_test(target, size):
+    """
+    The unit testing function. It @target device that perform the test to, 
+    and requires the maximum bio sector @size of the device. 
+    It returns True if test succeeds otherwise False. 
+    """
+    tracer_proc = Popen(["./biosplitmerge.py", "-S"], stdout=PIPE)
+    sleep(5)
+
+    comm = "dd if=/dev/zero of=%s bs=%dK count=1" % (args.device, 8*args.size)
+    os.system(comm)
+
+    print("Write finished. Sending Ctrl-C to tracer.")
+    tracer_proc.send_signal(SIGINT)
+    sleep(5)
+    if tracer_proc.poll() is None:
+        print("Tracer not terminated. Kill the tracer.")
+        tracer_proc.kill()
+
+    pid = str(os.getpid())
+    count = 0
+    for line in tracer_proc.stdout.readlines():
+        if line.split()[3] == pid:
+            count += 1
+    return count == 7
```

Comment:
`assert count == 7`?
 then we don't need the if check, we just run, and if it succeeds, we're good, if it fails, the process errors

Reply:
Makes sense.